### PR TITLE
Store: More Detailed Type Mismatch Error

### PR DESCRIPTION
**Pull request by ax3l**
Improve the error message for mismatched datatype and data passed to `store_chunk` by including both as well.

### CODE REVIEW

File: include/openPMD/RecordComponent.hpp [hunk 1]
```diff
@@ -174,7 +175,15 @@ RecordComponent::storeChunk(Offset o, Extent e, std::shared_ptr<T> data)
         throw std::runtime_error("Unallocated pointer passed during chunk store.");
     Datatype dtype = determineDatatype(data);
     if( dtype != getDatatype() )
-        throw std::runtime_error("Datatypes of chunk and dataset do not match.");
+    {
+        std::ostringstream oss;
+        oss << "Datatypes of chunk ("
+            << getDatatype()
```
Comment (ax3l) [inline include/openPMD/RecordComponent.hpp]: @C0nsultant did I mix up the two? is `dtype` of the incoming data the current `chunk` or the `dataset`? ;-)

Comment (ax3l) [reply include/openPMD/RecordComponent.hpp]: yep, I think I mixed it up. will also rename `of chunk` to `of chunk data`

Comment (anokfireball) [reply include/openPMD/RecordComponent.hpp]: [```getDatatype()```](https://github.com/openPMD/openPMD-api/blob/0.4.0-alpha/src/backend/BaseRecordComponent.cpp#L47-L51) is equal to the datatype assigned to a dataset/```RecordComponent``` through [```resetDataset()```](https://github.com/openPMD/openPMD-api/blob/0.4.0-alpha/src/RecordComponent.cpp#L46-L60).
So yes, it is actually the wrong way around.

Comment (ax3l) [reply include/openPMD/RecordComponent.hpp]: thx, already fixed locally. will push the rebase when #324 is in later today :)

File: CHANGELOG.rst [hunk 2]
```diff
@@ -30,6 +30,7 @@ Other
 """""
 
 - CMake: invasive tests not enabled by default #323
+- ``store_chunk``: more detailed detailed type mismatch error #322
```
Comment (ax3l) [inline CHANGELOG.rst]: double `detailed` -.-

File: include/openPMD/RecordComponent.hpp (hunk 1 above)
Comment (ax3l) [reply include/openPMD/RecordComponent.hpp]: updated :)

File: test/CoreTest.cpp [hunk 3]
```diff
@@ -567,6 +570,8 @@ TEST_CASE( "wrapper_test", "[core]" )
     REQUIRE(o.iterations[6].particles["electrons"].particlePatches["numParticles"][RecordComponent::SCALAR].m_chunks->size() == 1);
     REQUIRE(pp["numParticles"][RecordComponent::SCALAR].m_chunks->size() == 1);
 #endif
+    REQUIRE_THROWS_WITH(o.iterations[6].particles["electrons"].particlePatches["numParticles"][RecordComponent::SCALAR].store(idx+1, 42.),
+                        Catch::Equals("Datatypes of chunk data (DOUBLE) and dataset (UINT64) do not match."));
```
Comment (anokfireball) [inline test/CoreTest.cpp:574]: Kudos for also including unit tests!

Comment (ax3l) [reply test/CoreTest.cpp:574]: sneakily improving coverage, one corner case at a time :)
